### PR TITLE
Async Improvements

### PR DESCRIPTION
**Pull request by MrNukealizer**
Added async versions of each endpoint. It may have butchered the readability of some files though...
Moved the Accept and User-Agent header setup to class initialization to avoid proliferating duplicate headers.
Things should behave a little better when requesting different scopes than what the refresh token had, at least when taking time to verify the token.

### CODE REVIEW

**Comment by wranders** [comment]
Interesting. I like it.
I don't think it butchers readability much at all, really. I do have some questions, though.

Since the async methods are being called directly in your changes, are there any paths left that still use the old methods? (I didn't see any.) If not, then there's really no need to keep the old ones.

I'll annotate my other question as a Review.

File: ESISharp/Web/EsiRequest.cs [hunk 1]
```diff
@@ -26,35 +26,39 @@ internal EsiRequest(ESIEve EsiObject, string RequestPath)
...
         {
-            var ArgString = Utils.ConstructUrlArgs(QueryArguments);
-            var ArgumentRequest = string.Concat(RequestUrl, ArgString);
-            var Response = GetAsync(ArgumentRequest);
+            var Response = GetAsync(QueryArguments);
             return Response.Result;
         }
 
-        private async Task<string> GetAsync(string Url)
+        internal async Task<string> GetAsync(object QueryArguments = null)
```
Comment (wranders) [inline ESISharp/Web/EsiRequest.cs]: Can you remove the default parameter and overload a parameter-less method? I like to avoid defaults where I can.

**Comment by MrNukealizer** [comment]
Done. Also I forgot I had started converting List\<T\> parameters of public methods to IEnumerable\<T\> for people who like to use arrays or other types of collections. I finished that up so it should all be consistent.

**Comment by wranders** [comment]
I noticed the IEnumerable changes. Good call.

Excellent work. Merging.

**Pull request merged**